### PR TITLE
[kvutils] Add config key to read set [KVL-746]

### PR DESCRIPTION
**Pull request by miklos-da**
Summary of changes:
* Make retrieval of the current configuration go through `CommitContext` to get accessed keys properly recorded.
* Improved test coverage of `Committer.getCurrentConfiguration`.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala [hunk 1]
```diff
@@ -191,6 +192,15 @@ class TransactionCommitterSpec extends WordSpec with Matchers with MockitoSugar
         }
       }
     }
+
+    "mark config key as accessed in context" in {
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala:196]: This is a good unit test but I would expect this change to have a higher-level effect on the pre-execution result. Is there a test case we can also add to `CommitterSpec`?

**Review by gerolf-da — APPROVED** [review]
👍

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala (hunk 1 above)
Comment (miklos-da) [reply ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala:196]: Yes, for `Committer` I have added tests covering how the current configuration is being retrieved.

Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala:196]: OK, awesome.

**Pull request merged**